bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -260,6 +267,33 @@ impl Room {
         Ok(Timeline::new(timeline))
     }
 
+    pub async fn media_events_timeline(
+        &self,
+        filter: MediaEventsTimelineFilter,
+    ) -> Result<Arc<Timeline>, ClientError> {
+        let room = &self.inner;
+
+        let mut builder = matrix_sdk_ui::timeline::Timeline::builder(room);
+
+        builder = builder.event_filter(move |event, _| match event {
```

Comment:
Heavily inspired by the [default_event_filter](https://github.com/matrix-org/matrix-rust-sdk/blob/main/crates/matrix-sdk-ui/src/timeline/controller/mod.rs#L182)

Reply:
Any chance you could reuse `TimelineEventTypeFilter` that already exists at the FFI layer? Instead of passing it to `RoomListItem::init_timeline`, we could have a new way to create another (non-default) timeline from a filter and a prefix, maybe.

Reply:
Ah, good shout, didn't realise that's a thing. Let me have a play 👍